src/beast/beast/http/basic_parser.h

```diff
@@ -90,20 +107,23 @@ class basic_parser
 public:
     using error_code = boost::system::error_code;
 
-    virtual
-    ~basic_parser() = default;
```

Comment:
I'm unconcerned about cases where callers destroy a derived class through the base